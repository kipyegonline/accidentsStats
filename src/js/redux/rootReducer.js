const initState = {
  ntsa: [],
  ntsb: [],
  loadedA: false,
  loadedB: false,
  iclass: [],
  auth: { isLoggedIn: true, userName: "kipyegonline" },
};

function rootReducer(state = initState, action) {
  if (action.type === "ADD_NTSA") {
    return { ...state, ntsa: action.payload, loadedA: true };
  } else if (action.type === "ADD_NTSB") {
    return { ...state, ntsb: action.payload, loadedB: true };
  } else if (action.type === "ADD_CLASS") {
    return { ...state, iclass: action.payload };
  } else {
    return state;
  }
}
export default rootReducer;
