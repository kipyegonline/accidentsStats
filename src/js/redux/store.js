import { createStore } from "redux";
import rootReducer from "./rootReducer";
const devtools = window.__REDUX_DEVTOOLS_EXTENSION__
  ? window.__REDUX_DEVTOOLS_EXTENSION__()
  : (f) => f;
export const store = createStore(rootReducer, devtools);
