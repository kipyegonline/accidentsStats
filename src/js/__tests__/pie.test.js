import React from "react";
import Enzyme, { mount, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import NTSAPieChart from "../components/Pie/piechart";
import { MockStore } from "./redux.test";
Enzyme.configure({ adapter: new Adapter() });
const data = [
  {
    fatalties: 40,
    slightlyInjured: 120,
    victimClass: "pedestrians",
    seriouslyInjured: 180,
    addedon: "2020-03-20",
  },
  {
    fatalties: 70,
    slightlyInjured: 200,
    victimClass: "passengers",
    seriouslyInjured: 280,
    addedon: "2020-04-20",
  },

  {
    fatalties: 140,
    slightlyInjured: 240,
    victimClass: "drivers",
    seriouslyInjured: 380,
    addedon: "2020-04-20",
  },
];
describe("<NTSAPie/>", () => {
  let wrapper = mount(
    <MockStore>
      <NTSAPieChart datas={data} a="fatalties" b="victimClass" />
    </MockStore>
  );

  describe("SVG", () => {
    it("counts paths", () => expect(wrapper.find("path")).toHaveLength(3));
    it("counts groups", () => expect(wrapper.find("g")).toHaveLength(4));
    it("counts text tags", () => expect(wrapper.find("text")).toHaveLength(8));
    it("counts rects", () => expect(wrapper.find("rect")).toHaveLength(3));
    it("checks text at 0", () =>
      expect(
        wrapper
          .find("text")
          .at(1)
          .text()
      ).toBe("Total:300"));
  });
  test("mouseenter event", () => {
    const wedge = wrapper.find("path").at(0);
    wedge.simulate("mouseenter", { target: { value: 0 } });
  });
});
