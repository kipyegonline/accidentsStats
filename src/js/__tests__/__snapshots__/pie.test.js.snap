// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NTSAPie/> matches snapshot 1`] = `
<svg
  className=""
  height={380}
  stroke="none"
  strokeWidth={1}
  style={
    Object {
      "background": "lightblue",
    }
  }
  width={500}
>
  <Text
    TA=""
    classlist=""
    fw="normal"
    fz="1.25rem"
    transformer={0}
    x={150}
    y={30}
  >
    <text
      className=""
      fontSize="1.25rem"
      fontWeight="normal"
      pointerEvents="none"
      textAnchor=""
      transform="translate(0)"
      x={150}
      y={30}
    >
      FATALTIES
       
      -
       
      March
      , 
      2020
    </text>
  </Text>
  <Group
    classlist=""
    fill="none"
    x={300}
    y={70}
  >
    <g
      className=""
      fill="none"
      shapeRendering="auto"
      transform="translate(300,70)"
    >
      <Rect
        classlist=""
        fill="black"
        hc={[Function]}
        height={10}
        hh={[Function]}
        hl={[Function]}
        key="0"
        opacity={1}
        stroke="none"
        strokew={1}
        style={Object {}}
        width={30}
        x={20}
        y={30}
      >
        <rect
          className=""
          fill="black"
          height={10}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          opacity={1}
          shapeRendering="crispEdges"
          stroke="none"
          strokeWidth={1}
          style={Object {}}
          width={30}
          x={20}
          y={30}
        />
      </Rect>
      <Rect
        classlist=""
        fill="black"
        hc={[Function]}
        height={10}
        hh={[Function]}
        hl={[Function]}
        key="1"
        opacity={1}
        stroke="none"
        strokew={1}
        style={Object {}}
        width={30}
        x={20}
        y={45}
      >
        <rect
          className=""
          fill="black"
          height={10}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          opacity={1}
          shapeRendering="crispEdges"
          stroke="none"
          strokeWidth={1}
          style={Object {}}
          width={30}
          x={20}
          y={45}
        />
      </Rect>
      <Rect
        classlist=""
        fill="black"
        hc={[Function]}
        height={10}
        hh={[Function]}
        hl={[Function]}
        key="2"
        opacity={1}
        stroke="none"
        strokew={1}
        style={Object {}}
        width={30}
        x={20}
        y={60}
      >
        <rect
          className=""
          fill="black"
          height={10}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          opacity={1}
          shapeRendering="crispEdges"
          stroke="none"
          strokeWidth={1}
          style={Object {}}
          width={30}
          x={20}
          y={60}
        />
      </Rect>
      <Text
        TA=""
        classlist=""
        fill="black"
        fw="normal"
        fz="1rem"
        height={10}
        key="0"
        transformer={0}
        x={55}
        y={40}
      >
        <text
          className=""
          fill="black"
          fontSize="1rem"
          fontWeight="normal"
          pointerEvents="none"
          textAnchor=""
          transform="translate(0)"
          x={55}
          y={40}
        >
          pedestrians
        </text>
      </Text>
      <Text
        TA=""
        classlist=""
        fill="black"
        fw="normal"
        fz="1rem"
        height={10}
        key="1"
        transformer={0}
        x={55}
        y={55}
      >
        <text
          className=""
          fill="black"
          fontSize="1rem"
          fontWeight="normal"
          pointerEvents="none"
          textAnchor=""
          transform="translate(0)"
          x={55}
          y={55}
        >
          passengers
        </text>
      </Text>
      <Text
        TA=""
        classlist=""
        fill="black"
        fw="normal"
        fz="1rem"
        height={10}
        key="2"
        transformer={0}
        x={55}
        y={70}
      >
        <text
          className=""
          fill="black"
          fontSize="1rem"
          fontWeight="normal"
          pointerEvents="none"
          textAnchor=""
          transform="translate(0)"
          x={55}
          y={70}
        >
          drivers
        </text>
      </Text>
    </g>
  </Group>
  &gt;
  <Group
    classlist=""
    fill="none"
    key="0"
    x={180}
    y={200}
  >
    <g
      className=""
      fill="none"
      shapeRendering="auto"
      transform="translate(180,200)"
    >
      <Path
        classlist=""
        d="M7.34788079488412e-15,-120A120,120,0,0,1,101.31935106024183,-64.29921539747959L0,0Z"
        fill="black"
        hh={[Function]}
        hl={[Function]}
        opacity={1}
        stroke=""
        strokew={1}
      >
        <path
          className=""
          d="M7.34788079488412e-15,-120A120,120,0,0,1,101.31935106024183,-64.29921539747959L0,0Z"
          fill="black"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          opacity={1}
          shapeRendering="auto"
          stroke=""
          strokeWidth={1}
          style={
            Object {
              "position": "relative",
              "transition": "all .25s ease-in",
            }
          }
        />
      </Path>
      <Text
        TA="end"
        classlist="pie-slices"
        fill="black"
        fw="normal"
        fz=""
        key="0"
        transformer={
          Array [
            28.905220446102923,
            -52.57840080263181,
          ]
        }
      >
        <text
          className="pie-slices"
          fill="black"
          fontSize=""
          fontWeight="normal"
          pointerEvents="none"
          textAnchor="end"
          transform="translate(28.905220446102923,-52.57840080263181)"
        >
          40
        </text>
      </Text>
    </g>
  </Group>
  <Group
    classlist=""
    fill="none"
    key="1"
    x={180}
    y={200}
  >
    <g
      className=""
      fill="none"
      shapeRendering="auto"
      transform="translate(180,200)"
    >
      <Path
        classlist=""
        d="M101.31935106024183,-64.29921539747959A120,120,0,0,1,44.174946322161325,111.5731783065902L0,0Z"
        fill="black"
        hh={[Function]}
        hl={[Function]}
        opacity={1}
        stroke=""
        strokew={1}
      >
        <path
          className=""
          d="M101.31935106024183,-64.29921539747959A120,120,0,0,1,44.174946322161325,111.5731783065902L0,0Z"
          fill="black"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          opacity={1}
          shapeRendering="auto"
          stroke=""
          strokeWidth={1}
          style={
            Object {
              "position": "relative",
              "transition": "all .25s ease-in",
            }
          }
        />
      </Path>
      <Text
        TA="end"
        classlist="pie-slices"
        fill="black"
        fw="normal"
        fz=""
        key="1"
        transformer={
          Array [
            57.06339097770921,
            18.541019662496858,
          ]
        }
      >
        <text
          className="pie-slices"
          fill="black"
          fontSize=""
          fontWeight="normal"
          pointerEvents="none"
          textAnchor="end"
          transform="translate(57.06339097770921,18.541019662496858)"
        >
          70
        </text>
      </Text>
    </g>
  </Group>
  <Group
    classlist=""
    fill="none"
    key="2"
    x={180}
    y={200}
  >
    <g
      className=""
      fill="none"
      shapeRendering="auto"
      transform="translate(180,200)"
    >
      <Path
        classlist=""
        d="M44.174946322161325,111.5731783065902A120,120,0,1,1,-2.2043642384652355e-14,-120L0,0Z"
        fill="black"
        hh={[Function]}
        hl={[Function]}
        opacity={1}
        stroke=""
        strokew={1}
      >
        <path
          className=""
          d="M44.174946322161325,111.5731783065902A120,120,0,1,1,-2.2043642384652355e-14,-120L0,0Z"
          fill="black"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          opacity={1}
          shapeRendering="auto"
          stroke=""
          strokeWidth={1}
          style={
            Object {
              "position": "relative",
              "transition": "all .25s ease-in",
            }
          }
        />
      </Path>
      <Text
        TA="end"
        classlist="pie-slices"
        fill="black"
        fw="normal"
        fz=""
        key="2"
        transformer={
          Array [
            -58.93723504372132,
            11.242878875143475,
          ]
        }
      >
        <text
          className="pie-slices"
          fill="black"
          fontSize=""
          fontWeight="normal"
          pointerEvents="none"
          textAnchor="end"
          transform="translate(-58.93723504372132,11.242878875143475)"
        >
          140
        </text>
      </Text>
    </g>
  </Group>
  <Text
    TA=""
    classlist=""
    fw="bold"
    fz=""
    transformer={0}
    x={280}
    y={360}
  >
    <text
      className=""
      fontSize=""
      fontWeight="bold"
      pointerEvents="none"
      textAnchor=""
      transform="translate(0)"
      x={280}
      y={360}
    >
      Total 
      fatalties
      : 
      250
    </text>
  </Text>
</svg>
`;
